packages/core/src/envelope.ts

```diff
@@ -101,12 +109,30 @@ export function createEventEnvelope(
   // TODO: This is NOT part of the hack - DO NOT DELETE
   delete event.sdkProcessingMetadata;
 
-  const envelopeHeaders = {
+  let envelopeHeaders: EventEnvelopeHeaders = {
     event_id: event.event_id as string,
     sent_at: new Date().toISOString(),
     ...(sdkInfo && { sdk: sdkInfo }),
     ...(!!tunnel && { dsn: dsnToString(dsn) }),
   };
+
+  if (eventType === 'transaction') {
+    const baggage = createBaggage(
+      dropUndefinedKeys({
+        environment: event.environment,
+        release: event.release,
+        transaction: event.transaction,
+        userid: event.user && event.user.id,
+        // user.segment currently doesn't exist explicitly in interface User (just as a record key)
+        usersegment: event.user && event.user.segment,
+      } as BaggageObj),
+    );
+    envelopeHeaders = {
+      ...envelopeHeaders,
+      ...(Object.keys(baggage[0]).length > 0 && { baggage: serializeBaggage(baggage) }),
```

Comment:
let's pull checking if baggage is empty obj into a helper so that we don't expose baggage DS internals?

Reply:
Good idea - done!